### ISSUE TITLE
Learning rate and weight decay factors applied twice during weight decay

### ISSUE THREAD

**Issue by drehero** [dholzmueller/pytabkit#40]
Hi,

when looking at the optimizer code I noticed that the lr and wd factors are applied twice during weight decay.

When the optimizer is stepped weight decay is applied like this:

```python
with torch.no_grad():
    for i, group in enumerate(self.opt.param_groups):
        wd = self.get_hyper_values('wd', i)
        lr = self.get_hyper_values('lr', i)
        if wd != 0.0:
            for p in group['params']:
                p.mul_(1.0 - wd * lr * p.hyper_factors.get('wd', 1.0) * p.hyper_factors.get('lr', 1.0))
```

However wd and lr are already multiplied by their respective factors when calling the `get_hyper_values` method:

```python
def get_hyper_values(self, name, i, use_hyper_factor=True):
        value = self.hyper_getters[name][i]()
        param = self.opt.param_groups[i]['params'][0]  # should only be one param
        if use_hyper_factor and name in param.hyper_factors:
            value *= param.hyper_factors[name]
        return value
```

So effective shrink coefficient is:

```
base_lr * base_wd * lr_factor**2 * wd_factor**2
```

instead of

```
base_lr * base_wd * lr_factor * wd_factor
```

For example:

```
plr_lr_factor=0.1, plr_wd_factor=1.0 -> WD becomes 0.01x instead of 0.1x
scale_lr_factor=6.0, scale_wd_factor=1.0 -> WD becomes 36x instead of 6x
```

Is this intended or should the `get_hyper_values` method be called with `use_hyper_factor=False`?

Relevant code:
https://github.com/dholzmueller/pytabkit/blob/main/pytabkit/models/optim/optimizers.py#L31-L62

**Comment by dholzmueller** [comment]
Thanks for raising the issue! I'm not sure how to best fix it while keeping things backward-compatible, but it's good to keep in mind for a future release and maybe I'll put a warning in the comments or so.
